### PR TITLE
Move chrono-tz arithmetic tests to integration

### PR DESCRIPTION
**Pull request by tustvold**
# Which issue does this PR close?

Closes #.

# Rationale for this change
 
Removes the feature from the sub-crate, as it only gates tests

# What changes are included in this PR?

# Are there any user-facing changes?

No, the arrow-arith feature has not been released

### CODE REVIEW

File: arrow-arith/Cargo.toml [hunk 1]
```diff
@@ -46,4 +46,3 @@ num = { version = "0.4", default-features = false, features = ["std"] }
 
 [features]
 simd = ["arrow-array/simd"]
-chrono-tz = ["arrow-array/chrono-tz"]
```
Comment (alamb) [inline arrow-arith/Cargo.toml:49]: isn't this technically a breaking change for anyone using https://crates.io/crates/arrow-arith directly with this feature? I doubt anyone will notice, but I do think we should flag the PR as API change

File: arrow/Cargo.toml [hunk 2]
```diff
@@ -300,3 +300,7 @@ required-features = ["chrono-tz", "prettyprint"]
 [[test]]
 name = "timezone"
 required-features = ["chrono-tz"]
+
+[[test]]
+name = "arithmetic"
+required-features = ["chrono-tz"]
```
Comment (alamb) [inline arrow/Cargo.toml:306]: arithmetic is a fairly general name that only has timezone related tests now -- do you plan to move more tests there over time?

**Review by alamb — APPROVED** [review]
Nice cleanup - thank you @tustvold

File: arrow-arith/Cargo.toml (hunk 1 above)
Comment (tustvold) [reply arrow-arith/Cargo.toml:49]: It hasn't been released yet

File: arrow/Cargo.toml (hunk 2 above)
Comment (tustvold) [reply arrow/Cargo.toml:306]: Yes

**Pull request merged**